geoips/filenames/base_paths.py

```diff
@@ -1,252 +1,171 @@
     )
-elif getenv("GEOIPS_ANCILDAT_AUTOGEN"):
-    PATHS["GEOIPS_ANCILDAT_AUTOGEN"] = getenv("GEOIPS_ANCILDAT_AUTOGEN").rstrip("/")
-
-if not getenv("LOGDIR"):
-    PATHS["LOGDIR"] = pathjoin(PATHS["GEOIPS_OUTDIRS"], "logs")
-else:
-    PATHS["LOGDIR"] = getenv("LOGDIR").rstrip("/")
-
-if not getenv("GEOIPSDATA"):
-    PATHS["GEOIPSDATA"] = pathjoin(PATHS["GEOIPS_OUTDIRS"], "geoipsdata")
-else:
-    PATHS["GEOIPSDATA"] = getenv("GEOIPSDATA").rstrip("/")
-
-if not getenv("GEOIPS_VERS"):
-    PATHS["GEOIPS_VERS"] = "0.0.0"
-else:
-    PATHS["GEOIPS_VERS"] = getenv("GEOIPS_VERS")
-
-
-if getenv("TCWWW"):
-    PATHS["TCWWW"] = getenv("TCWWW").rstrip("/")
-else:
-    PATHS["TCWWW"] = pathjoin(PATHS["GEOIPS_OUTDIRS"], "preprocessed", "tcwww")
-
-if getenv("TCWWW_URL"):
-    PATHS["TCWWW_URL"] = getenv("TCWWW_URL").rstrip("/")
-else:
-    PATHS["TCWWW_URL"] = PATHS["TCWWW"]
-
-if getenv("PUBLICWWW"):
-    PATHS["PUBLICWWW"] = getenv("PUBLICWWW").rstrip("/")
-else:
-    PATHS["PUBLICWWW"] = pathjoin(PATHS["GEOIPS_OUTDIRS"], "preprocessed", "publicwww")
-
-if getenv("PUBLICWWW_URL"):
-    PATHS["PUBLICWWW_URL"] = getenv("PUBLICWWW_URL").rstrip("/")
-else:
-    PATHS["PUBLICWWW_URL"] = PATHS["PUBLICWWW"]
-
-if getenv("PRIVATEWWW"):
-    PATHS["PRIVATEWWW"] = getenv("PRIVATEWWW").rstrip("/")
-else:
-    PATHS["PRIVATEWWW"] = pathjoin(
-        PATHS["GEOIPS_OUTDIRS"], "preprocessed", "privatewww"
+    paths["GEOIPS_BASEDIR"] = get_env_var(
+        "GEOIPS_BASEDIR",
+        os.path.abspath(os.path.join(paths["GEOIPS_PACKAGES_DIR"], os.pardir)),
     )
 
-if getenv("PRIVATEWWW_URL"):
-    PATHS["PRIVATEWWW_URL"] = getenv("PRIVATEWWW_URL").rstrip("/")
-else:
-    PATHS["PRIVATEWWW_URL"] = PATHS["PRIVATEWWW"]
-
-PATHS["TC_DECKS_DB"] = pathjoin(
-    PATHS["GEOIPS_OUTDIRS"], "longterm_files", "tc", "tc_decks.db"
-)
-if getenv("TC_DECKS_DB"):
-    PATHS["TC_DECKS_DB"] = getenv("TC_DECKS_DB")
-
-if getenv("TC_DECKS_DIR"):
-    PATHS["TC_DECKS_DIR"] = getenv("TC_DECKS_DIR").rstrip("/")
-else:
-    PATHS["TC_DECKS_DIR"] = pathjoin(
-        PATHS["GEOIPS_OUTDIRS"], "longterm_files", "tc", "decks"
-    )
+    geoips_global_variables = {
+        # GeoIPS Documentation URL
+        "GEOIPS_DOCS_URL": r"https://nrlmmd-geoips.github.io/geoips/",
+        # Version
+        "GEOIPS_VERS": os.getenv("GEOIPS_VERS", "0.0.0"),
+        # Operational User
+        "GEOIPS_OPERATIONAL_USER": os.getenv("GEOIPS_OPERATIONAL_USER", False),
+        # Copyright Information
+        "GEOIPS_COPYRIGHT": os.getenv("GEOIPS_COPYRIGHT", "NRL-Monterey"),
+        "GEOIPS_COPYRIGHT_ABBREVIATED": os.getenv(
+            "GEOIPS_COPYRIGHT_ABBREVIATED", "NRLMRY"
+        ),
+        # Configuration and Queue
+        "GEOIPS_RCFILE": os.getenv("GEOIPS_RCFILE", ""),
+        "DEFAULT_QUEUE": os.getenv("DEFAULT_QUEUE", None),
+        # Computer Identifier
+        "BOXNAME": socket.gethostname(),
+    }
+
+    paths.update(geoips_global_variables)
+
+    derivative_directory_paths = {
```

Comment:
Maybe name this derivative_directory_paths_defaults , and leave a comment that this is specifically the defaults for any path based environment variables,  which all default to locations under either BASE_PATHS or GEOIPS_OUTDIRS, to keep all geoips code and outputs consolidated unless explicitly specified through environment variables.  Also that eventually the environment variable overrides will be replaced (or supplemented) with a yaml or json config.

Reply:
Maybe these should also use "get_env_var" instead of specifying the defaults directly in the dict, for consistency and future-proofing for when we are also pulling from the YAML/JSON config?

Reply:
Added! I put some of this info in the module docstring since it applies to the whole file.